ledgers/ccf/transaction_processor/pdo_tp.cpp

```diff
@@ -364,6 +370,18 @@ namespace ccfapp
                         HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Enclave attestation report verification Failed. Invalid user report data");
                 }
 
+                // Verify 64-bit enclave
+                if((reportBody->attributes.flags & SGX_FLAGS_MODE64BIT) == 0)
+                    return ccf::make_error(HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput,
+                        "Enclave attestation report verification Failed. Enclave is not 64-bit");
+
+                // Verify SGX debug flag
+                std::string flag = std::to_string((reportBody->attributes.flags & SGX_FLAGS_DEBUG) > 0);
+                if(flag != expected_sgx_measurements.sgx_debug_flag)
+                    return ccf::make_error(HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput,
+                        "Enclave attestation report verification Failed. Enclave debug flag " + flag + 
+                        " does not match policy " + expected_sgx_measurements.sgx_debug_flag);
+
```

Comment:
More of a philosophical question: Why would we check attestation if it is sgx debug mode ON? What is the security model?

Reply:
Because that's the case when you want to test/debug PDO in HW-mode, and you can have legitimate attestations in that case.